### PR TITLE
Create current environemnt link when --print-deps-only has no env

### PR DESCRIPTION
**Pull request by hroncok**
(Better reviewed commit by commit.)

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -195,3 +195,24 @@ def test_regular_after_deps_only():
...
     assert "six" in result.stdout
+
+
+def test_print_deps_without_python_command(tmp_path):
+    bin = tmp_path / "bin"
+    bin.mkdir()
+    tox_link = bin / "tox"
+    tox_path = shutil.which("tox")
+    tox_link.symlink_to(tox_path)
+    env = {**os.environ, "PATH": str(bin)}
```
Comment (encukou) [inline tests/test_integration.py]: Shouldn't this prepend to PATH, rather than replace it?

**Comment by encukou** [comment]
With this, after you run `rm -r .tox; tox --print-deps-only`, regular `tox` seems to consider the environment created and will try to install dependencies into the current environment.

File: tests/test_integration.py (hunk 1 above)
Comment (hroncok) [reply tests/test_integration.py]: We want a PATH without `python`. But we need the `tox` command to work. Hence we create custom PATH that contains `tox` only. Previous PATH already contains `tox`.

**Comment by hroncok** [comment]
> regular tox seems to consider the environment created and will try to install dependencies into the current environment.

I'll update the "unsupported" check.

**Comment by hroncok** [comment]
I've amended tests and docs, but I think the situation you are describing is not happening. Could you please give full reproducer?

**Comment by hroncok** [comment]
In other words:

After you run `rm -r .tox; tox --print-deps-only`, regular `tox` aborts with "not supported". At least that is what I believe `test_regular_after_first_deps_only_is_not_supported` is checking.

**Comment by hroncok** [comment]
Oh, it aborts after it installs the package :(

**Comment by hroncok** [comment]
Added a test that finds out about this.

**Comment by hroncok** [comment]
Fixed.

Also, in the future, we could simply always delete the fake environment in `tox_cleanup` hook and get rid of this restriction, but I'll leave that to another PR.

**Comment by hroncok** [comment]
Opened https://github.com/fedora-python/tox-current-env/issues/14 as a followup.

**Pull request merged**